core/src/main/java/org/apache/calcite/rex/RexRangeRef.java

```diff
@@ -57,6 +57,7 @@
       int offset) {
     this.type = rangeType;
     this.offset = offset;
+    this.digest = "offset(" + offset + ")";
   }
 
```

Comment:
This class is used only during the process of translating a SqlNode to RelNode. So it is hard to add a test.